src/Slist.hs

```diff
@@ -672,6 +673,21 @@ init sl@Slist{..} = case sSize of
     _        -> Slist (P.init sList) (sSize - 1)
 {-# INLINE init #-}
 
+{- | @O(1)@. 'cons' is 'Slist' analogue to ':' for lists.
+It adds the given element to the beginning of the list.
+
+>>> cons 'a' $ one 'b'
+Slist {sList = "ab", sSize = Size 2}
+
+@
+>> __'cons' 0 $ 'infiniteSlist' [1..]__
+Slist {sList = [0..], sSize = 'Infinity'}
+@
+-}
+cons :: a -> Slist a -> Slist a
+cons x (Slist xs s) = Slist (x:xs) $ s + 1
+{-# INLINE cons #-}
```

Comment:
Do you think it would make sense to write strict `cons'` as well in the spirit of #36?